sdcm/data_operations/data_operation.py

```diff
@@ -0,0 +1,293 @@
+            try:
+                partition_values = self.get_unique_partition_values(partitions_amount=partitions_amount_to_delete)
+                where_condition = self.build_where_condition(columns=self.schema_helpers.partition_keys,
+                                                             values=partition_values)
+                self.log.debug("Where condition for delete: %s", where_condition)
+                self.delete_partition(where_condition_to_delete=where_condition)
+                time.sleep(self.interval)
+                self.validate_deleted_rows_not_resurrected(where_condition_to_delete=where_condition)
+            except Exception as error:  # pylint: disable=broad-except
+                self.log.error(traceback.format_exc())
+                case_event.severity = Severity.CRITICAL if isinstance(error, ResurrectionDataFound) else Severity.ERROR
+                case_event.message = str(error)
+
+    def delete_half_partition(self, partitions_amount_to_delete: int = 1):
+        with DataOperationsCaseEvent(case="DeleteHalfPartition") as case_event:
+            try:
+                partition_values = self.get_unique_partition_values(partitions_amount=partitions_amount_to_delete)
+                where_condition = self.build_where_condition(columns=self.schema_helpers.partition_keys,
+                                                             values=partition_values)
+                max_clustering_values = self.get_max_clustering_value_in_partition(where_condition=where_condition)
+                if not max_clustering_values:
+                    return
+
+                where_condition_to_delete = []
+                for pk_clause, clustering_value in max_clustering_values.items():
+                    where_condition_to_delete.append(f" {pk_clause} and "
+                                                     f"{clustering_value['ck_name']} > {int(clustering_value['value'] / 2)}")
+
+                self.log.debug("Where condition for delete: %s", where_condition_to_delete)
+
+                self.delete_partition(where_condition_to_delete=where_condition_to_delete)
+                time.sleep(self.interval)
+                self.validate_deleted_rows_not_resurrected(where_condition_to_delete=where_condition_to_delete)
+                return
+            except Exception as error:  # pylint: disable=broad-except
+                self.log.error(traceback.format_exc())
+                case_event.severity = Severity.CRITICAL if isinstance(error, ResurrectionDataFound) else Severity.ERROR
+                case_event.message = str(error)
+                return
+
+
+# pylint: disable=too-many-instance-attributes
+class DataOperationThread(Thread):
+    """
+    Run thread with data manipulations during entire test. Sleep between operations.
+    For now one operation is supported: delete full partition
+    """
+    DEFAULT_INTERVAL_BETWEEN_OPERATIONS_SEC = 900
```

Comment:
I think this should be configurable, in some cases we might want it more often ?